modules/machineset-creating-non-guaranteed-instances.adoc

```diff
@@ -1,24 +1,68 @@
 // Module included in the following assemblies:
 //
 // * machine_management/creating_machinesets/creating-machineset-aws.adoc
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc
+// * machine_management/creating_machinesets/creating-machineset-azure.adoc
+
+ifeval::["{context}" == "creating-machineset-aws"]
+:aws:
+endif::[]
+ifeval::["{context}" == "creating-machineset-azure"]
+:azure:
+endif::[]
+ifeval::["{context}" == "creating-machineset-gcp"]
+:gcp:
+endif::[]
 
 [id="machineset-creating-non-guaranteed-instance_{context}"]
-= Creating Spot Instances
+ifdef::aws[= Creating Spot Instances]
+ifdef::azure[= Creating Spot VMs]
+ifdef::gcp[= Creating preemptible VM instances]
 
-You can launch a Spot Instance on AWS by adding SpotMarketOptions to your MachineSet YAML file.
+ifdef::aws[You can launch a Spot Instance on AWS by adding SpotMarketOptions to your MachineSet YAML file.]
+ifdef::azure[You can launch a Spot VM on Azure by adding SpotVMOptions to your MachineSet YAML file.]
+ifdef::gcp[You can launch a preemptible VM instance on GCP by adding the `Preemptible` field to your MachineSet YAML file.]
 
 .Procedure
 * Add the following line under the `providerSpec` field:
 +
+ifdef::aws[]
 [source,yaml]
 ----
 providerSpec:
   value:
     spotMarketOptions: {} <1>
 ----
-<1> You can optionally set the `maxPrice` for the `spotMarketOptions` field to limit the cost of the Spot Instance, for example `maxPrice: '2.50'`. If the `maxPrice` is set, this value is used as the hourly maximum spot price. If not set, the maximum price defaults to charge up to the On-Demand Instance price.
+<1> You can optionally set the `maxPrice` for the `spotMarketOptions` field to limit the cost of the Spot Instance, for example `maxPrice: '2.50'`.
+If the `maxPrice` is set, this value is used as the hourly maximum spot price. If not set, the maximum price defaults to charge up to the On-Demand Instance price.
 
 [NOTE]
 ====
 It is strongly recommended to use the default On-Demand price as the `maxPrice` value and to not set the maximum price for Spot Instances.
 ====
+endif::aws[]
+ifdef::azure[]
+[source,yaml]
+----
+providerSpec:
+  value:
+    spotVMOptions: {} <1>
+----
+<1> You can optionally set the `maxPrice` for the `spotVMOptions` field to limit the cost of the Spot VM, for example `maxPrice: '0.98765'`.
+If the `maxPrice` is set, this value is used as the hourly maximum spot price. If not set, the maximum price defaults to `-1` and charges the current price, up to the standard VM price.
+Azure caps Spot VM prices at the standard price and therefore will not evict an instance set with the default `maxPrice`.
```

Comment:
This should say something along the lines of `not evict and instance due to pricing` as the instance may still be evicted due to capacity restrictions

Reply:
@JoelSpeed I applied your feedback to the callout 👍  Thank you!